types.cc

```diff
@@ -67,14 +70,19 @@ requires requires {
 sstring
 time_point_to_string(const T& tp)
 {
-    int64_t count = tp.time_since_epoch().count();
-    auto time = boost::posix_time::from_time_t(0) + boost::posix_time::milliseconds(count);
-    constexpr std::string_view units = "milliseconds";
-  try {
-    return boost::posix_time::to_iso_extended_string(time);
-  } catch (const std::exception& e) {
-    return format("{} {} ({})", count, units, e.what());
-  }
+    auto d = std::div(int64_t(tp.time_since_epoch().count()), int64_t(1000));
+    std::time_t seconds = d.quot;
+    std::tm tm = *gmtime(&seconds);
```

Comment:
Do you know if gmtime uses a per-thread variable or uses the same one for all threads? The manual doesn't claim it uses a per-thread variable.

So please change to use gmtime_r.